### PR TITLE
HPCC-21518 DALI - Make sure audience and message class is sensible

### PR DESCRIPTION
**Pull request by shamser**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by shamser** [comment]
@jakesmith Please can your review

File: dali/datest/dfuwutest.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ void testAbort(const char *wuid)
         wu->requestAbort();
     }
     else
-        ERRLOG("WUID %s not found", wuid);
+        OERRLOG("WUID %s not found", wuid);
```
Comment (jakesmith) [inline dali/datest/dfuwutest.cpp]: UERRLOG

File: dali/datest/dfuwutest.cpp [hunk 2]
```diff
@@ -421,7 +421,7 @@ IFileDescriptor *createRoxieFileDescriptor(const char *cluster, const char *lfn,
         grpname.append("__servers");
     Owned<IGroup> grp = queryNamedGroupStore().lookup(grpname.str());
     if (!grp) {
-        ERRLOG("Logical group %s not found",grpname.str());
+        OERRLOG("Logical group %s not found",grpname.str());
```
Comment (jakesmith) [inline dali/datest/dfuwutest.cpp]: UERRLOG

File: dali/datest/dfuwutest.cpp [hunk 3]
```diff
@@ -432,7 +432,7 @@ IFileDescriptor *createRoxieFileDescriptor(const char *cluster, const char *lfn,
         dirxpath.appendf("Software/RoxieCluster[@name=\"%s\"]/Roxie%sProcess[@channel=\"%d\"]/@dataDirectory",cluster,servers?"Server":"Slave",i+1);
         const char * dir = envroot->queryProp(dirxpath.str());
         if (!dir) {
-            ERRLOG("dataDirectory not specified");
+            OERRLOG("dataDirectory not specified");
```
Comment (jakesmith) [inline dali/datest/dfuwutest.cpp]: UERRLOG

File: dali/datest/floodtest.cpp [hunk 4]
```diff
@@ -367,7 +367,7 @@ void nodeInfo::sendError(const char *errstr1,const char *errstr2)
     StringBuffer err(errstr1);
     if (errstr2)
         err.append(" ").append(errstr2);
-    ERRLOG("%s",err.str());
+    OERRLOG("%s",err.str());
```
Comment (jakesmith) [inline dali/datest/floodtest.cpp]: UERRLOG

File: dali/datest/floodtest.cpp [hunk 5]
```diff
@@ -415,12 +415,12 @@ void floodtestClient(const char *grpname,unsigned ngb)
             PROGLOG("Started");
         }
         else {
-            ERRLOG("Failed to find node in group");
+            OERRLOG("Failed to find node in group");
```
Comment (jakesmith) [inline dali/datest/floodtest.cpp]: UERRLOG

File: dali/datest/floodtest.cpp [hunk 6]
```diff
@@ -415,12 +415,12 @@ void floodtestClient(const char *grpname,unsigned ngb)
...
             mb.append(FLOODTEST_ERROR).append(myrank).append("Failed to find node in group");
         }
     }
     else {
-        ERRLOG("Failed to find node in group");
+        OERRLOG("Failed to find node in group");
```
Comment (jakesmith) [inline dali/datest/floodtest.cpp]: UERRLOG

File: dali/datest/floodtest.cpp [hunk 7]
```diff
@@ -663,7 +663,7 @@ bool runSlaves(IGroup *grp,const char *progname,const char *daliserver,const cha
                 else {
                     StringBuffer ips;
                     ip.getText(ips);
-                    ERRLOG("Failed to run slave on %d (%s)",idx+1,ips.str());
+                    OERRLOG("Failed to run slave on %d (%s)",idx+1,ips.str());
```
Comment (jakesmith) [inline dali/datest/floodtest.cpp]: UERRLOG

File: dali/datest/floodtest.cpp [hunk 8]
```diff
@@ -793,7 +793,7 @@ void floodtestServer(const char *exename,const char *daliserver,const char *grpn
     bool loopback = (ngb==0);
     Owned<IGroup> group = queryNamedGroupStore().lookup(grpname);
     if (!group) {
-        ERRLOG("Cannot find group %s",grpname);
+        OERRLOG("Cannot find group %s",grpname);
```
Comment (jakesmith) [inline dali/datest/floodtest.cpp]: UERRLOG

File: dali/daunittest/daunittest.cpp [hunk 9]
```diff
@@ -69,7 +69,7 @@ class DfsTestProgressListener : public CPPUNIT_NS::TestListener
         CPPUNIT_NS::Exception *e = failure.thrownException();
         if (e)
             s.appendf(" %s %s",e->message().shortDescription().c_str(),e->message().details().c_str());
-        ERRLOG("%s",s.str());
+        OERRLOG("%s",s.str());
```
Comment (jakesmith) [inline dali/daunittest/daunittest.cpp]: UERRLOG

File: dali/dfu/dfurun.cpp [hunk 10]
```diff
@@ -1117,12 +1117,12 @@ class CDFUengine: public CInterface, implements IDFUengine
             progress->setState(DFUstate_aborted);
             /* no break */
         case DFUstate_aborted:
-            WARNLOG("DFURUN: Workunit %s aborted",dfuwuid);
+            OWARNLOG("DFURUN: Workunit %s aborted",dfuwuid);
```
Comment (jakesmith) [inline dali/dfu/dfurun.cpp]: IWARNLOG

File: dali/sasha/packetstore.cpp [hunk 11]
```diff
@@ -240,7 +240,7 @@ class CDataPacketTransaction: extends CInterface
     {
         CriticalBlock block(store.lockingsect);
         if (locked) {
-            LOG(MCerror, unknownJob, "Warning: transaction previously locked (lockRead)");
+			DBGLOG("Warning: transaction previously locked (lockRead)");
```
Comment (jakesmith) [inline dali/sasha/packetstore.cpp]: odd indentation

File: dali/server/daserver.cpp [hunk 12]
```diff
@@ -185,7 +185,7 @@ int main(int argc, char* argv[])
             fileMsgHandler = lf->beginLogging();
         }
 
-        DBGLOG("Build %s", BUILD_TAG);
+        LOG(MCoperatorInfo, unknownJob, "Build %s", BUILD_TAG);
```
Comment (jakesmith) [inline dali/server/daserver.cpp]: PROGLOG

File: dali/server/daserver.cpp [hunk 13]
```diff
@@ -432,7 +432,7 @@ int main(int argc, char* argv[])
             stopPerformanceMonitor();
             throw;
         }
-        PrintLog("DASERVER[%d] starting - listening to port %d",myrank,queryMyNode()->endpoint().port);
+        DBGLOG("DASERVER[%d] starting - listening to port %d",myrank,queryMyNode()->endpoint().port);
```
Comment (jakesmith) [inline dali/server/daserver.cpp]: PROGLOG

File: fs/dafilesrv/dafilesrv.cpp [hunk 14]
```diff
@@ -280,7 +280,7 @@ bool uninstallService(const char *servicename,const char *servicedisplayname)
                 if ( ss.dwCurrentState == SERVICE_STOPPED ) 
                     PROGLOG("%s Service stopped",servicedisplayname); 
                 else 
-                    ERRLOG("%s failed to stop",servicedisplayname); 
+                    oERRLOG("%s failed to stop",servicedisplayname);
```
Comment (jakesmith) [inline fs/dafilesrv/dafilesrv.cpp]: typo' - won't compile under windows.

OERRLOG

File: fs/dafscontrol/dafscontrol.cpp [hunk 15]
```diff
@@ -262,7 +262,7 @@ unsigned applyNodes(const char *grpip, ApplyMode mode, unsigned ver, bool isdali
             StringBuffer epstr;
             ForEachItemIn(i,result) {
                 result.item(i).getUrlStr(epstr.clear());
-                ERRLOG("%s: %s not running DAFILESRV",grpip,epstr.str());
+                OERRLOG("%s: %s not running DAFILESRV",grpip,epstr.str());
```
Comment (jakesmith) [inline fs/dafscontrol/dafscontrol.cpp]: UERRLOG

File: fs/dafscontrol/dafscontrol.cpp [hunk 16]
```diff
@@ -226,9 +226,9 @@ unsigned applyNodes(const char *grpip, ApplyMode mode, unsigned ver, bool isdali
                 result.item(i).getUrlStr(epstr.clear());
                 StringAttrItem &attr = resultstr.item(i);
                 if (attr.text.length()==0) 
-                    ERRLOG("%s: %s not running DAFILESRV",grpip,epstr.str());
+                    OERRLOG("%s: %s not running DAFILESRV",grpip,epstr.str());
                 else 
-                    ERRLOG("%s: %s %s running DAFILESRV version %s",grpip,(mode==AMstopver)?"was":"is",epstr.str(),attr.text.get());
+                    OERRLOG("%s: %s %s running DAFILESRV version %s",grpip,(mode==AMstopver)?"was":"is",epstr.str(),attr.text.get());
```
Comment (jakesmith) [inline fs/dafscontrol/dafscontrol.cpp]: UERRLOG

File: fs/dafscontrol/dafscontrol.cpp [hunk 17]
```diff
@@ -226,9 +226,9 @@ unsigned applyNodes(const char *grpip, ApplyMode mode, unsigned ver, bool isdali
                 result.item(i).getUrlStr(epstr.clear());
                 StringAttrItem &attr = resultstr.item(i);
                 if (attr.text.length()==0) 
-                    ERRLOG("%s: %s not running DAFILESRV",grpip,epstr.str());
+                    OERRLOG("%s: %s not running DAFILESRV",grpip,epstr.str());
```
Comment (jakesmith) [inline fs/dafscontrol/dafscontrol.cpp]: UERRLOG

File: fs/dafscontrol/dafscontrol.cpp [hunk 18]
```diff
@@ -193,7 +193,7 @@ unsigned applyNodes(const char *grpip, ApplyMode mode, unsigned ver, bool isdali
                     {
                         unsigned err = stopRemoteServer(sock);
                         if (err!=0) {
-                            ERRLOG("Could not stop server on %s, %d returned",epstr.str(),err);
+                            OERRLOG("Could not stop server on %s, %d returned",epstr.str(),err);
```
Comment (jakesmith) [inline fs/dafscontrol/dafscontrol.cpp]: UERRLOG

File: fs/dafscontrol/dafscontrol.cpp [hunk 19]
```diff
@@ -110,7 +110,7 @@ unsigned applyNodes(const char *grpip, ApplyMode mode, unsigned ver, bool isdali
     if (!isdali||!getCluster(grpip,eps)) {
         SocketEndpoint ep(grpip);
         if (ep.isNull()) {
-            ERRLOG("%s is not a group name or ip",grpip);
+            OERRLOG("%s is not a group name or ip",grpip);
```
Comment (jakesmith) [inline fs/dafscontrol/dafscontrol.cpp]: UERRLOG

**Review by jakesmith — CHANGES_REQUESTED** [review]

**Comment by jakesmith** [comment]
@shamser - back to you.

1 general comment - for command line tools (that may be used by an operator) (e.g. daliadmin), the messages that intended to be seen on the CLI should be user, not operator.

File: dali/base/dasds.cpp [hunk 20]
```diff
@@ -3866,12 +3866,12 @@ int CSDSTransactionServer::run()
                     try { coven.reply(mb); }
                     catch (IJSOCK_Exception *e)
                     {
-                        LOG(MCwarning, unknownJob, e, "Failed to reply to client (CSDSTransactionServer thread)");
+                        LOG(MCinternalError, unknownJob, e, "Failed to reply to client (CSDSTransactionServer thread)");
```
[the remark this answers is not in the document]
Comment (shamser) [reply dali/base/dasds.cpp]: IWARNLOG does not have a version that supports exceptions... I'll have to use LOG(MCinternalWarning,...) for the same purpose.

File: dali/base/dasds.cpp [hunk 21]
```diff
@@ -4811,15 +4811,15 @@ IPropertyTree *loadStore(const char *storeFilename, IPTreeMaker *iMaker, unsigne
...
+            IWARNLOG("Error processing store %s - CRC ERROR (file size=%" I64F "d, validation crc=%x, calculated crc=%x)", storeFilename, iFileIOStore->size(), crcValidation, crc); // not fatal yet (maybe later)
     }
     catch (IException *e)
     {
         if (!abort)
         {
             StringBuffer s("Exception - loading store file : ");
             s.appendf("%s", storeFilename);
-            LOG(MCoperatorError, unknownJob, e, s.str());
+            LOG(MCinternalError, unknownJob, e, s.str());
```
[the remark this answers is not in the document]
Comment (shamser) [reply dali/base/dasds.cpp]: LOG(MCinternalError,..) should be the same as IERRLOG but with support for the IException

File: dali/base/dasds.cpp [hunk 22]
```diff
@@ -5521,7 +5521,7 @@ class CStoreHelper : implements IStoreHelper, public CInterface
             }
             catch (IException *e)
             {
-                LOG(MCoperatorError, unknownJob, e, "Exception(1) - Error saving store file");
+                LOG(MCinternalError, unknownJob, e, "Exception(1) - Error saving store file");
```
[the remark this answers is not in the document]
Comment (shamser) [reply dali/base/dasds.cpp]: "LOG(MCoperatorError," is equivalent of OERRLOG but support IException parameter

File: dali/base/dasds.cpp (hunk 20 above)
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dasds.cpp]: >IWARNLOG does not have a version that supports exceptions

may be worth introducing one.

File: dali/base/dasds.cpp (hunk 21 above)
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dasds.cpp]: I assumed an IERRLOG existed that took a IException.
Feels like there should be one to make usage more consistent.

File: dali/base/dasds.cpp (hunk 22 above)
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dasds.cpp]: from jlog.hpp: 

>inline IException *OERRLOG(IException *except, const char *prefix=NULL)

already seems to be a OERRLOG that supports an exception

[the remark this answers is not in the document]
Comment (shamser) [reply dali/base/dasds.cpp]: you're right.  I missed that.

File: dali/base/dasds.cpp (hunk 21 above)
[the remark this answers is not in the document]
Comment (shamser) [reply dali/base/dasds.cpp]: There is an IERRLOG for IExceptions.  Didn't see that initially.  Changing as suggested.

**Comment by shamser** [comment]
@jakesmith That was a lot of changes!  I've modified as suggested.

File: system/jlib/jlog.hpp [hunk 23]
```diff
@@ -988,6 +988,24 @@ inline void IWARNLOG(LogMsgCode code, char const * format, ...)
     va_end(args);
 }
 
+inline IException *IWARNLOG(IException *except, const char *prefix=NULL)
```
Comment (jakesmith) [inline system/jlib/jlog.hpp]: trivial: nullptr preferable

**Comment by jakesmith** [comment]
@shamser - one trivial comment (re. nullptr).
Please squash.

**Comment by shamser** [comment]
@jakesmith nullptr change implemented and squashed

File: dali/base/dafdesc.cpp [hunk 24]
```diff
@@ -2955,7 +2955,7 @@ IFileDescriptor *createFileDescriptorFromRoxieXML(IPropertyTree *tree,const char
                 }
             }
             else
-                ERRLOG("createFileDescriptorFromRoxie: %s missing part %s",id,xpath.str());
+                OERRLOG("createFileDescriptorFromRoxie: %s missing part %s",id,xpath.str());
```
Comment (jakesmith) [inline dali/base/dafdesc.cpp]: think should be changed to IERRLOG

File: dali/base/dasds.cpp [hunk 25]
```diff
@@ -509,7 +509,7 @@ class CSubscriberContainerBase : public CInterfaceOf<IInterface>
         }
         catch (IException *e)
         {
-            LOG(MCuserWarning, e, "SDS: Error notifying subscriber");
+            LOG(MCinternalError, e, "SDS: Error notifying subscriber");
```
Comment (jakesmith) [inline dali/base/dasds.cpp]: could be IERRLOG

File: dali/base/dasds.cpp [hunk 26]
```diff
@@ -992,19 +992,19 @@ void writeDelta(StringBuffer &xml, IFile &iFile, const char *msg="", unsigned re
         {
             exception.setown(e);
             StringBuffer s(msg);
-            LOG(MCoperatorError, unknownJob, e, s.append("writeDelta, failed").str());
+            LOG(MCinternalError, unknownJob, e, s.append("writeDelta, failed").str());
```
Comment (jakesmith) [inline dali/base/dasds.cpp]: could be IERRLOG

File: dali/base/dasds.cpp [hunk 27]
```diff
@@ -1309,7 +1309,7 @@ class CBackupHandler : public CInterface, implements IThreaded
         }
         catch (IException *e)
         {
-            LOG(MCoperatorWarning, e, "BackupHandler(async) write operation failed, possible backup data loss");
+            LOG(MCoperatorWarning, unknownJob, e, "BackupHandler(async) write operation failed, possible backup data loss");
```
Comment (jakesmith) [inline dali/base/dasds.cpp]: could be IERRLOG

File: dali/base/dasds.cpp [hunk 28]
```diff
@@ -7163,19 +7163,19 @@ void CCovenSDSManager::stop()
 
 void CCovenSDSManager::restart(IException * e)
 {
-    LOG(MCwarning, unknownJob, "-------: stopping SDS server");
+    DBGLOG("-------: stopping SDS server");
```
Comment (jakesmith) [inline dali/base/dasds.cpp]: Some of this should be PROGLOG so we see something when DBGLOG's are suppressed.
Make this one a PROGLOG.

File: dali/base/dasds.cpp [hunk 29]
```diff
@@ -7163,19 +7163,19 @@ void CCovenSDSManager::stop()
...
+    DBGLOG("-------: stopped");
+    DBGLOG("-------: saving current store . . . . . .");
     saveStore();
-    LOG(MCwarning, unknownJob, "-------: store saved.");
-    LOG(MCwarning, unknownJob, "-------: restarting SDS server restart");
+    DBGLOG("-------: store saved.");
+    DBGLOG("-------: restarting SDS server restart");
     start();
-    LOG(MCwarning, unknownJob, "-------: restarted");
+    DBGLOG("-------: restarted");
```
Comment (jakesmith) [inline dali/base/dasds.cpp]: and this one

File: dali/base/dasds.cpp [hunk 30]
```diff
@@ -8639,7 +8639,7 @@ bool CCovenSDSManager::fireException(IException *e)
 {
     //This code is rather dodgy (and causes more problems than it solves) 
     // so ignore unhandled exceptions for the moment! 
-    LOG(MCoperatorError, unknownJob, e, "Caught unhandled exception!");
+    LOG(MCinternalError, unknownJob, e, "Caught unhandled exception!");
```
Comment (jakesmith) [inline dali/base/dasds.cpp]: could be IERRLOG

File: dali/base/dasds.cpp [hunk 31]
```diff
@@ -8649,7 +8649,7 @@ bool CCovenSDSManager::fireException(IException *e)
                 LOG(MCdisaster, unknownJob, e, "FATAL, too many exceptions");
                 return false; // did not successfully handle.
             }
-            LOG(MCoperatorError, unknownJob, e, "Exception while restarting or shutting down");
+            LOG(MCinternalError, unknownJob, e, "Exception while restarting or shutting down");
```
Comment (jakesmith) [inline dali/base/dasds.cpp]: could be IERRLOG

File: dali/datest/datest.cpp [hunk 32]
```diff
@@ -958,7 +958,7 @@ void TestRemoteFile(unsigned part,unsigned of)
 
 void QTest(bool testput)
 {
-    PrintLog("starting QTest %s",testput?"put":"get");
+    DBGLOG("starting QTest %s",testput?"put":"get");
```
Comment (jakesmith) [inline dali/datest/datest.cpp]: Not sure about any (most) of the changes to DBGLOG in this file.
We prob. want to see datest test tracing even in release, so wouldn't want these to go away if we start suppressing DBGLOG.

Bettter to change them all to either PROGLOG I think.

File: dali/sasha/packetstore.cpp [hunk 33]
```diff
@@ -240,7 +240,7 @@ class CDataPacketTransaction: extends CInterface
     {
         CriticalBlock block(store.lockingsect);
         if (locked) {
-            LOG(MCerror, unknownJob, "Warning: transaction previously locked (lockRead)");
+            DBGLOG("Warning: transaction previously locked (lockRead)");
```
Comment (jakesmith) [inline dali/sasha/packetstore.cpp]: these and others below that were LOG(MCerror, ..
look like they should be IERRLOG

**Comment by shamser** [comment]
@jakesmith Modified as you've suggested.  Shall I squash?

**Comment by jakesmith** [comment]
@shamser - yes please go ahead and squash.

**Comment by shamser** [comment]
@jakesmith squashed.

**Comment by jakesmith** [comment]
@richardkchapman - ready to merge.

**Pull request merged**